app/controllers/application_controller.rb

```diff
@@ -1,12 +1,11 @@
 class ApplicationController < ActionController::Base
+  before_action :authenticate_user!
   before_action :configure_permitted_parameters, if: :devise_controller?
-  before_action :set_current_user
 
   protected
 
   def configure_permitted_parameters
-    devise_parameter_sanitizer.permit(:sign_up, keys: %i[name posts_counter])
-    devise_parameter_sanitizer.permit(:account_update, keys: %i[name posts_counter])
+    devise_parameter_sanitizer.permit(:sign_up, keys: %i[email name password password_confirmation posts_counter])
   end
 
   def set_current_user
```

Comment:
- [ ] Nice adding `devise` and nice remove set_current_user from before action but it would be better to remove the method also cause you don't need it Devise offers it 
 - also to  follow `YAGNI` principle

Reply:
**Hi @youmari Thanks for the review in this case we do need it because removing it would give an error like the screenshot below please let us know if you have a solution for it thanks.**
<img width="1582" alt="Screenshot 2023-05-17 at 1 31 51 AM" src="https://github.com/anas599/blog/assets/90115863/7630aa72-a9b9-4c0e-90de-248d3c80aafa">

Reply:
@anas599 I think you solved the prob